File: UISegmentedControlObjC/UISegmentedControlObjC/PieView.m
//
//  PieView.m
//  UISegmentedControlObjC
//
//  Created by Hugo Flores Perez on 3/25/20.
//  Copyright © 2020 Hugo Flores. All rights reserved.
//

#import "PieView.h"

@interface ShapeDrawer ()
@property UIBezierPath *path;
- createCircle;
@end

@implementation PieView

- (void)drawRect:(CGRect)rect {
    [self createCircle];
    
    [UIColor.orangeColor setFill];
    [self.path fill];
    
    [UIColor.purpleColor setStroke];
    [self.path stroke];
}

@end

File: UISegmentedControlSwift/UISegmentedControlSwift/PieView.swift
//
//  PieView.swift
//  UISegmentedControlSwift
//
//  Created by Field Employee on 3/25/20.
//  Copyright © 2020 Hugo Flores. All rights reserved.
//

import UIKit

class PieView: Demoview {

    // Only override draw() if you perform custom drawing.
    // An empty implementation adversely affects performance during animation.
    override func draw(_ rect: CGRect) {
        self.createPie()
        
        // Specify the fill color and apply it to the path.
        UIColor.orange.setFill()
        path.fill()
        
        // Specify a border (stroke) color.
        UIColor.purple.setStroke()
        path.stroke()
    }

}
